nemo/collections/asr/data/audio_to_text_dali.py

```diff
@@ -0,0 +1,404 @@
+                        audio = audio + self.dither * gaussian_noise
+
+                    # Preemphasis filter
+                    if self.preemph > 0.0:
+                        audio = dali.fn.preemphasis_filter(audio, preemph_coeff=self.preemph)
+
+                    # Power spectrogram
+                    spec = dali.fn.spectrogram(
+                        audio,
+                        nfft=self.n_fft,
+                        window_length=self.window_size,
+                        window_step=self.window_stride
+                    )
+
+                    if feature_type == 'mel_spectrogram' or feature_type == 'mfcc':
+                        # Spectrogram to Mel Spectrogram
+                        spec = dali.fn.mel_filter_bank(
+                            spec, sample_rate=self.sample_rate, nfilter=self.n_mels, normalize=True,
+                            freq_low=self.freq_low, freq_high=self.freq_high
+                        )
+                        # Mel Spectrogram to MFCC
+                        if feature_type == 'mfcc':
+                            spec = dali.fn.mfcc(spec, n_mfcc=self.n_mfcc)
+
+                    # Logarithm
+                    if self.log_zero_guard_type == 'add':
+                        spec = spec + self.log_zero_guard_value
+
+                    spec = dali.fn.to_decibels(
+                        spec, multiplier=math.log(10), reference=1.0, cutoff_db=math.log(self.log_zero_guard_value)
+                    )
+
+                    # Normalization
+                    spec = dali.fn.normalize(spec, axes=self.normalization_axes)
+
+                    # Extracting the length of the spectrogram
+                    shape_start = dali.types.Constant(np.array([1], dtype=np.float32), device='cpu')
+                    shape_len = dali.types.Constant(np.array([1], dtype=np.float32), device='cpu')
+                    spec_len = dali.fn.slice(
+                        dali.fn.shapes(spec), shape_start, shape_len, normalized_anchor=False, normalized_shape=False, axes=(0,)
+                    )
+
+                    # Pads feature dimension to be a multiple of `pad_to` and the temporal dimension to be as big as the largest sample (shape -1)
+                    spec = dali.fn.pad(
+                        spec, fill_value=self.pad_value, axes=(0, 1), align=(self.pad_to, 1), shape=(1, -1)
+                    )
+
+                    pipe.set_outputs(spec, spec_len, transcript, transcript_len)
+
+            # Building DALI pipeline
+            pipe.build()
+
+        if has_preprocessor:
+            output_names = ['processed_signal', 'processed_signal_len', 'transcript_raw', 'transcript_raw_len']
+        else:
+            output_names = ['audio', 'audio_len', 'transcript_raw', 'transcript_raw_len']
+
+        self._iter = DALIPytorchIterator(
+            self.pipes, output_map=output_names, reader_name="Reader",
+            fill_last_batch=True, dynamic_shape=True, auto_reset=True
```

Comment:
Please use newly introduced `last_batch_policy=LastBatchPolicy.FILL`.
BTW - is that what we want or we should set `pad_last_batch` in the reader and set the policy to PARTIAL here.